neuro_san/internals/chat/data_driven_chat_session.py

```diff
@@ -41,17 +43,18 @@ class DataDrivenChatSession:
     in using data-driven agent tool graphs.
     """
 
-    def __init__(self, registry: AgentToolRegistry):
+    def __init__(self, agent_network: AgentNetwork):
         """
         Constructor
 
-        :param registry: The AgentToolRegistry to use.
+        :param agent_network: The AgentNetwork to use.
         """
-        # This block contains top candidates for state storage that needs to be
-        # retained when session_ids go away.
-        self.front_man: FrontMan = None
+        # We make a copy of the AgentNetwork at this level to allow for interactions
+        # within this scope to modify the network topology.
+        agent_network_copy: AgentNetwork = copy.deepcopy(agent_network)
+        self.registry: AgentToolRegistry = AgentToolRegistry(agent_network_copy)
```

Comment:
2 small but significant changes here to provide architectural cognitive relief:

1. Deep copy the data in the AgentNetwork so that later on any given request has leeway to modify topology if needed.

2. This is now the one place where an AgentNetwork is passed to an AgentToolRegistry constructor.
That is, the pairing of config with ActivationFactory capabilities now starts at this *internal* level, not at the service/session level as done previously.